### PR TITLE
Http2 files v4

### PR DESCRIPTION
**Pull request by catenacyber**
Right test condition for http2 files
Adds a check that there is no anomaly in http2 basic test
check for file decompression
adding a check for brotli decompressed files

Modifies #375 with adding per-filter features check
This changes the framework

### CODE REVIEW

**Comment by catenacyber** [comment]
@inashivb could you review especially the commit `framework: adds feature filtering per check`?
Thanks

File: tests/http2-bugfixes/test.yaml [hunk 1]
```diff
@@ -14,3 +14,17 @@ checks:
       count: 0
       match:
         event_type: anomaly
+# check gzip decompresser
+  - filter:
+      feature: HTTP2_DECOMPRESSION
```
Comment (jasonish) [inline tests/http2-bugfixes/test.yaml:19]: Can you add an example of this to the main readme?

**Review by inashivb — COMMENTED** [review]
LGTM Philippe.

File: tests/http2-bugfixes/test.yaml (hunk 1 above)
Comment (catenacyber) [reply tests/http2-bugfixes/test.yaml:19]: @inashivb this is not exactly the same format as in https://redmine.openinfosecfoundation.org/issues/4185
Does it matter ?

(will update the README after this answer)

Comment (inashivb) [reply tests/http2-bugfixes/test.yaml:19]: @catenacyber no this is OK. feature is a good key name here.

**Comment by catenacyber** [comment]
Replaced by #398

File: tests/http2-bugfixes/test.yaml (hunk 1 above)
Comment (catenacyber) [reply tests/http2-bugfixes/test.yaml:19]: Added the example in README in next PR #398

**Pull request closed without merging**